daemon/src/file_asset_source.rs

```diff
@@ -385,7 +437,7 @@ impl FileAssetSource {
                 txn.put_bytes(self.tables.asset_id_to_path, &asset, &dst)?;
             }
             if let Some(existing_metadata) = existing_metadata {
-                self.delete_metadata(txn, &evt.src)?;
+                self.delete_metadata(txn, &evt.src);
                 txn.put(self.tables.path_to_metadata, &dst, &existing_metadata)?;
```

Comment:
maybe expect here too, and remove the Result return? Should be no transient errors in this function since it operates on the DB.